src/engine/plugins/match.cpp

```diff
@@ -229,6 +240,29 @@ Status MatchPlugin::HandleRequest(const RoutingAlgorithmsInterface &algorithms,
         return Error("NoMatch", "Could not match the trace.", json_result);
     }
 
+    // trace was split, we don't support the waypoints parameter across multiple match objects
+    if (sub_matchings.size() > 1 && !parameters.waypoints.empty())
+    {
+        return Error("NoMatch", "Could not match the trace with the given waypoints.", json_result);
+    }
+
+    // Error: Check if user-supplied waypoints can be found in the resulting matches
+    {
+        std::set<std::size_t> tidied_waypoints(tidied.parameters.waypoints.begin(),
+                                               tidied.parameters.waypoints.end());
+        std::for_each(sub_matchings.begin(), sub_matchings.end(), [&](const SubMatching &sm) {
```

Comment:
Instead of `for_each` you can use `for (const auto& sm : sub_matchings) { for (auto index : sm.indices) {} }` which looks a little cleaner.